### PR TITLE
adding model download and get github access token sdk

### PR DESCRIPTION
**Pull request by jwang98052**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ai/azure-ai-resources/azure/ai/resources/github/accesstoken/get_github_token.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# ---------------------------------------------------------
```
Comment (luigiw) [inline sdk/ai/azure-ai-resources/azure/ai/resources/github/accesstoken/get_github_token.py:1]: As we discussed offline. Please move both files to this folder in ML SDK. https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/ml/azure-ai-ml/azure/ai/ml/_utils

File: sdk/ai/azure-ai-resources/azure/ai/resources/github/accesstoken/get_github_token.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+def parse_string_to_dict(token_text):
+    pairs = token_text.split('&')
+    result_dict = {}
+    for pair in pairs:
+        key, value = pair.split('=')
+        result_dict[key] = value    
+    return result_dict
+
+    
+# Replace 'YOUR_CLIENT_ID' with your GitHub app client ID
```
Comment (luigiw) [inline sdk/ai/azure-ai-resources/azure/ai/resources/github/accesstoken/get_github_token.py:62]: Let's remove the test code. You can also add a dummy test case here to not loose the sample code. https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/ml/azure-ai-ml/tests

File: sdk/ai/azure-ai-resources/azure/ai/resources/github/modeldownload/download_model.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        if copied_file_path:
+            print(f"The blob has been downloaded to folder: {copied_file_path}")
+        else:
+            print("Failed to downloaded blob.")
+    except Exception as e:
+        print(f"Error getting blob: {e}")
+        return None
+    
+# Example usage
+if __name__ == "__main__":
```
Comment (luigiw) [inline sdk/ai/azure-ai-resources/azure/ai/resources/github/modeldownload/download_model.py:79]: Same here to remove test code and add a dummy test case.

File: sdk/ai/azure-ai-resources/azure/ai/resources/github/modeldownload/download_model.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            with open(local_file_path, "wb") as local_file:
+                download_stream = blob_client.download_blob()
+                local_file.write(download_stream.readall())
+            print(f"Blob '{blob_prop.name}' copied to '{local_file_path}' successfully.")
+        return local_folder
+    except Exception as e:
+        print(f"Error copying blob to local folder: {e}")
+        return None
+
+def download_model(model_asset_id, github_auth_token, local_destination_path, environment = 'test'):
```
Comment (luigiw) [inline sdk/ai/azure-ai-resources/azure/ai/resources/github/modeldownload/download_model.py:60]: As we discussed. Let's remove `environment`.

**Review by luigiw — CHANGES_REQUESTED** [review]

File: sdk/ai/azure-ai-resources/azure/ai/resources/github/accesstoken/get_github_token.py (hunk 1 above)
Comment (jwang98052) [reply sdk/ai/azure-ai-resources/azure/ai/resources/github/accesstoken/get_github_token.py]: moved

File: sdk/ai/azure-ai-resources/azure/ai/resources/github/accesstoken/get_github_token.py (hunk 2 above)
Comment (jwang98052) [reply sdk/ai/azure-ai-resources/azure/ai/resources/github/accesstoken/get_github_token.py]: The test is just a one line function call and the some parameters like client id and github token used in the test are not going to check in any way. So I just removed the test. I will document the sample calls and explain how to get client id in the doc.

File: sdk/ai/azure-ai-resources/azure/ai/resources/github/modeldownload/download_model.py (hunk 4 above)
Comment (jwang98052) [reply sdk/ai/azure-ai-resources/azure/ai/resources/github/modeldownload/download_model.py]: done

**Review by luigiw — APPROVED** [review]

**Pull request closed without merging**